infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -476,6 +476,10 @@ public void processLockAcquire() {
     public void processLockRelease() {
         log.debug("Lock released");
         sessionManager.getReplicationContext().setIsLeader(false);
+        if(logReplicationEventListener != null) {
+            logReplicationEventListener.stop();
+        }
+        sessionManager.stopClientConfigListener();
```

Comment:
this is already done from sessionManager.stopReplication(), which is invoked from discovery service's stopLogReplication() on L484.  Should we move the event listener stop over there as well?

Reply:
Ohh...I missed that. I'm moving this invocation from sessionManager.stopReplication to discoveryService.stopReplication so all the listeners are subscribed and unsubscribed together. Otherwise its easy to miss one of them.

I also moved them to "stopReplication()" because the listeners should be stopped whenever the replication is stopped and not just on lock release.